common.sh

```diff
@@ -157,9 +157,12 @@ function getrepo() {
 
   pushd "$SOURCES_DIR"
   if [ ! -d "$dirname" ]; then
-    git clone "$url" "$dirname"
+    mkdir "$dirname"
     pushd "$dirname"
-    git checkout "$rev"
+    git init
+    git remote add origin "$url"
+    git fetch --depth=1 origin "$rev" || git fetch origin
```

Comment:
why the fallback?

Reply:
Because SourceForge refuses to give a specific commit unless it is named (tag or branch). Sadly it is the one needing this most so I doubt whether this PR will give anything noticeable.

Anyway, why do you fetch specific commits directly from git and not use official release tarballs?

Reply:
I don't recall exactly. It didn't occur to me that git would be slower, so I had no reason to favor one or the other. I believe in at least one case the official tarball had an unstable hash, and in another case, the tarball didn't work with emscripten, while the the latest github version did.

On my next big update I'll incorporate this change, with some modifications. I need minetest, irrlicht, and webshims repos to be full checkouts since I develop those in tandem, so I'd need a toggle for that.